lib/season_stat_team.rb

```diff
@@ -0,0 +1,80 @@
+require_relative 'game_collection'
+require_relative 'game_team_collection'
+require_relative 'team_collection'
+
+class SeasonStatTeam
+
+  def initialize(game_team_file_path, team_file_path)
+    @game_team_collection = GameTeamCollection.new(game_team_file_path)
+    @team_collection = TeamCollection.new(team_file_path)
+  end
+
+  def get_season_game_teams(season)
+    @game_team_collection.game_team_list.find_all do |game_team|
+      game_team.game_id[0..3] == season[0..3]
+    end
+  end
+
+  def get_team_ids_by_season(season)
+    get_season_game_teams(season).map do |game_team|
+      game_team.team_id
+    end.uniq
+  end
+
+  def get_team_name(team_id)
+    team_name_by_id = @team_collection.teams_list.find do |team|
+      team.team_id == team_id
+    end
+    team_name_by_id.team_name
+  end
+
+  def get_goals_by_team_season(team_id, season)#refactor to a reduce
+    team_goals = 0
+    get_season_game_teams(season).each do |game_team|
+      if game_team.team_id.to_s == team_id
+        team_goals += game_team.goals
+      end
+    end
+    team_goals
+  end
+
+  def get_shots_by_team_season(team_id, season)
+    team_shots = 0
+    get_season_game_teams(season).each do |game_team|
+      if game_team.team_id.to_s == team_id
+        team_shots += game_team.shots
+      end
+    end
+    team_shots
+  end
+
+  def team_shots_to_goal_ratio_by_season(team_id, season)
+    total_goals = get_goals_by_team_season(team_id, season).to_f
+    total_shots = get_shots_by_team_season(team_id, season)
+    (total_shots / total_goals).round(3)
+  end
+
+  def create_team_data_by_season(season)
+    get_team_ids_by_season(season).reduce({}) do |acc, team_id|
+      acc[team_id.to_s] = {
+        team_name: get_team_name(team_id),
+        goal_ratio: team_shots_to_goal_ratio_by_season(team_id.to_s, season)
+      }
+      acc
+    end
+  end
+
```

Comment:
a lot of these methods are starting to look the same as others, so we'll need to refactor them once everything is complete

Reply:
They look very similar to some of mine, but I think an option would be a module at this point. If we were to combine methods.